cdap-ui/app/cdap/components/PreviewData/RecordView/RecordContainer.tsx

```diff
@@ -0,0 +1,147 @@
+    const recordNum = parseInt(newVal.split(' ')[1], 10);
+    setRecord(recordNum);
+  };
+
+  const showInputTabs = inputs.length > 1;
+  const showOutputTabs = outputs.length > 1;
+
+  const handleTabClick = (id) => {
+    setActiveTab(id);
+  };
+
+  const getTabConfig = (stagesInfo, recordNum: number) => {
+    const recordIndex = recordNum - 1;
+    const tabs = stagesInfo.map(([stageName, recordInfo], index) => {
+      return {
+        id: index + 1,
+        name: stageName,
+        content: (
+          <RecordTable headers={recordInfo.schemaFields} record={recordInfo.records[recordIndex]} />
+        ),
+        paneClassName: 'record-pane',
+      };
+    });
+    return {
+      defaultTab: 1,
+      layout: 'horizontal',
+      tabs,
+    };
+  };
+
+  const getTabs = (config) => {
+    return <ConfigurableTab tabConfig={config} onTabClick={handleTabClick} activeTab={activeTab} />;
+  };
+
+  return (
+    <div>
+      <RecordNavigator
+        selectedRecord={selectedRecord}
+        numRecords={numRecords}
+        updateRecord={updateRecord}
+        prevOperation={() => setRecord(selectedRecord - 1)}
+        nextOperation={() => setRecord(selectedRecord + 1)}
+      />
+      <div className={classes.outerContainer}>
+        <If condition={!selectedNode.isSource && !selectedNode.isCondition}>
+          <div
+            className={classnames(classes.innerContainer, {
+              [classes.split]: !selectedNode.isSource && !selectedNode.isSink,
+            })}
+          >
+            <h2 className={classes.h2Title}>{T.translate(`${I18N_PREFIX}.inputHeader`)}</h2>
+            {showInputTabs
+              ? getTabs(getTabConfig(inputs, selectedRecord))
+              : inputs.map(([stageName, stageInfo]) => {
+                  return (
+                    <RecordTable
+                      headers={stageInfo.schemaFields}
+                      record={stageInfo.records[selectedRecord - 1]}
+                    />
+                  );
+                })}
+          </div>
+        </If>
+        <If condition={!selectedNode.isSink && !selectedNode.isCondition}>
+          <div
+            className={classnames(classes.innerContainer, {
+              [classes.split]: !selectedNode.isSource && !selectedNode.isSink,
+            })}
+          >
+            <h2 className={classes.h2Title}>{T.translate(`${I18N_PREFIX}.outputHeader`)}</h2>
+            {showOutputTabs
+              ? getTabs(getTabConfig(outputs, selectedRecord))
+              : outputs.map(([stageName, stageInfo]) => {
+                  return (
+                    <RecordTable
+                      headers={stageInfo.schemaFields}
+                      record={stageInfo.records[selectedRecord - 1]}
+                    />
+                  );
+                })}
+          </div>
+        </If>
+        <If condition={!selectedNode.isSink && !selectedNode.isCondition && showOutputTabs}>
+          <div></div>
```

Comment:
is this empty intentional or should there be some message?

Reply:
This shouldn't be there at all - I've removed this If component.